### PR TITLE
AEAD limits

### LINKED ISSUE #203: Limits on # of records
* Should we mention limits on # of records per connection in TLS 1.2, similarly to Sec. 5.5 of RFC 8446?
* See https://datatracker.ietf.org/doc/draft-irtf-cfrg-aead-limits/02/ 
* Or even copy the numbers from 8446.
* DONE: Peter to ask Martin & Chris. The numbers for 1.2 would be the same as for QUIC and 1.3

### PR DESCRIPTION
**Pull request by yaronf**
Fixes #203.

I left a TODO for when the "limits" draft gets revised with the TLS 1.2 derivation.

### CODE REVIEW

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 1]
```diff
@@ -550,6 +550,23 @@ Note that some profiles of TLS 1.2 use different cipher suites. For example, {{?
...
+a Key Update) to rotate the session key.
+
+For all AES-GCM cipher suites recommended for TLS 1.2 in this document, the limit
+for one connection is 2^24.5 full-size records (about 24 million).
+This is the same number as for TLS 1.3 with the equivalent cipher suites.
+
+[[TODO: refer to {{?I-D.irtf-cfrg-aead-limits}} once they have added the derivation
+for TLS 1.2, which is different from TLS 1.3. Different derivation, same numbers.]]
+
+For all TLS 1.3 cipher suites, readers are referred to Sec. 5.5 of {{RFC8446}}.
```
Comment (thomas-fossati) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:568]:
```suggestion
For all TLS 1.3 cipher suites, readers are referred to Section 5.5 of {{RFC8446}}.
```

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 2]
```diff
@@ -550,6 +550,23 @@ Note that some profiles of TLS 1.2 use different cipher suites. For example, {{?
...
+## Limits on Key Usage
+
+All ciphers have an upper limit on the amount of traffic that can be securely encrypted
+with any given key. In the case of AEAD cipher suites, the limit is typically determined
+by the cipher's integrity guarantees. When the amount of traffic for a particular connection
+has reached the limit, an implementation SHOULD perform a new handshake (or in TLS 1.3,
+a Key Update) to rotate the session key.
+
+For all AES-GCM cipher suites recommended for TLS 1.2 in this document, the limit
+for one connection is 2^24.5 full-size records (about 24 million).
```
Comment (thomas-fossati) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:562]:
```suggestion
for one connection is 2<sup>24.5</sup> full-size records (about 24 million).
```

**Review by thomas-fossati — COMMENTED** [review]
LGTM

(I have left a couple of purely editorial comments, feel free to drop them)

**Review by thomas-fossati — APPROVED** [review]

File: BCP195bis/draft-ietf-uta-rfc7525bis.md [hunk 3]
```diff
@@ -550,6 +550,23 @@ Note that some profiles of TLS 1.2 use different cipher suites. For example, {{?
...
 
+## Limits on Key Usage
+
+All ciphers have an upper limit on the amount of traffic that can be securely encrypted
+with any given key. In the case of AEAD cipher suites, the limit is typically determined
+by the cipher's integrity guarantees. When the amount of traffic for a particular connection
+has reached the limit, an implementation SHOULD perform a new handshake (or in TLS 1.3,
+a Key Update) to rotate the session key.
+
+For all AES-GCM cipher suites recommended for TLS 1.2 in this document, the limit
```
Comment (stpeter) [inline BCP195bis/draft-ietf-uta-rfc7525bis.md:561]: s/recommend/recommended/

**Review by stpeter — APPROVED** [review]

File: BCP195bis/draft-ietf-uta-rfc7525bis.md (hunk 3 above)
Comment (yaronf) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:561]: Confused: it already says "recommended".

Comment (stpeter) [reply BCP195bis/draft-ietf-uta-rfc7525bis.md:561]: Oops, I misread it.

**Pull request merged**